### ISSUE TITLE
How do I read config props from resource/file only if it exists and otherwise ignore it?

### ISSUE THREAD

**Issue by deiga** [npryce/konfig#18]
```
val config = systemProperties() overriding
        EnvironmentVariables() overriding
        ConfigurationProperties.fromResource("local.properties")
```

Currently this fails when `local.properties` doesn't exist. Is there a way to let it ignore that file, if it's not there?

**Comment by nikbucher** [comment]
I had a similar use case, because I want to provide a config file in the user.dir only if the defaults are not suitable. So this issue is probably solved with #20.

**Comment by npryce** [comment]
This would have to be implemented differently for each source of configuration.  But Konfig now provides an `EmptyConfiguration` object which can be used in the else branch of a conditional expression that tests for the existence of a configuration source.

E.g. for a (fictitious) example configuration implementation:

`(if (registry.hasKey("my-app")) RegistryConfiguration(registry, "my-app") else EmptyConfiguration)`
